consensus/beacon/consensus.go

```diff
@@ -264,7 +264,13 @@ func (beacon *Beacon) verifyHeader(chain consensus.ChainHeaderReader, header, pa
 		return consensus.ErrInvalidNumber
 	}
 	// Verify the header's EIP-1559 attributes.
-	return misc.VerifyEip1559Header(chain.Config(), parent, header)
+	if err := misc.VerifyEip1559Header(chain.Config(), parent, header); err != nil {
+		return err
+	}
+	if !chain.Config().IsShanghai(header.Number) && header.WithdrawalHash != nil {
+		return fmt.Errorf("invalid withdrawalHash: have %s, expected nil", header.WithdrawalHash)
+	}
```

Comment:
Shouldn't it be 
- Verify `!shanghai -> header.withdrawalhash == nil` AND  `shanghai -> header.withdrawalhash != nil` ?

Reply:
I think you are correct, will update.